ProcessLib/HeatTransportBHE/BHE/BHE_1U.h

```diff
@@ -0,0 +1,213 @@
+              typename SingleUnknownMatrixType,
+              typename RMatrixType,
+              typename RPiSMatrixType,
+              typename RSMatrixType>
+    void assembleRMatrices(
+        int const idx_bhe_unknowns,
+        Eigen::MatrixBase<SingleUnknownMatrixType> const& matBHE_loc_R,
+        Eigen::MatrixBase<RMatrixType>& R_matrix,
+        Eigen::MatrixBase<RPiSMatrixType>& R_pi_s_matrix,
+        Eigen::MatrixBase<RSMatrixType>& R_s_matrix) const
+    {
+        switch (idx_bhe_unknowns)
+        {
+            case 0:  // PHI_fig
+                R_matrix.block(0, 2 * NPoints, NPoints, NPoints) +=
+                    -1.0 * matBHE_loc_R;
+                R_matrix.block(2 * NPoints, 0, NPoints, NPoints) +=
+                    -1.0 * matBHE_loc_R;
+
+                R_matrix.block(0, 0, NPoints, NPoints) +=
+                    1.0 * matBHE_loc_R;  // K_i1
+                R_matrix.block(2 * NPoints, 2 * NPoints, NPoints, NPoints) +=
+                    1.0 * matBHE_loc_R;  // K_ig
+                break;
+            case 1:  // PHI_fog
+                R_matrix.block(NPoints, 3 * NPoints, NPoints, NPoints) +=
+                    -1.0 * matBHE_loc_R;
+                R_matrix.block(3 * NPoints, NPoints, NPoints, NPoints) +=
+                    -1.0 * matBHE_loc_R;
+
+                R_matrix.block(NPoints, NPoints, NPoints, NPoints) +=
+                    1.0 * matBHE_loc_R;  // K_o1
+                R_matrix.block(3 * NPoints, 3 * NPoints, NPoints, NPoints) +=
+                    1.0 * matBHE_loc_R;  // K_og
+                break;
+            case 2:  // PHI_gg
+                R_matrix.block(2 * NPoints, 3 * NPoints, NPoints, NPoints) +=
+                    -1.0 * matBHE_loc_R;
+                R_matrix.block(3 * NPoints, 2 * NPoints, NPoints, NPoints) +=
+                    -1.0 * matBHE_loc_R;
+
+                R_matrix.block(2 * NPoints, 2 * NPoints, NPoints, NPoints) +=
+                    1.0 * matBHE_loc_R;  // K_ig  // notice we only have
+                                         // 1 PHI_gg term here.
+                R_matrix.block(3 * NPoints, 3 * NPoints, NPoints, NPoints) +=
+                    1.0 * matBHE_loc_R;  // K_og  // see Diersch 2013 FEFLOW
+                                         // book page 954 Table M.2
+                break;
+            case 3:  // PHI_gs
+                R_s_matrix.template block<NPoints, NPoints>(0, 0).noalias() +=
+                    1.0 * matBHE_loc_R;
+
+                R_pi_s_matrix.block(2 * NPoints, 0, NPoints, NPoints) +=
+                    -1.0 * matBHE_loc_R;
+                R_pi_s_matrix.block(3 * NPoints, 0, NPoints, NPoints) +=
+                    -1.0 * matBHE_loc_R;
+                R_matrix.block(2 * NPoints, 2 * NPoints, NPoints, NPoints) +=
+                    1.0 * matBHE_loc_R;  // K_ig
+                R_matrix.block(3 * NPoints, 3 * NPoints, NPoints, NPoints) +=
+                    1.0 * matBHE_loc_R;  // K_og
+                break;
```

Comment:
Why no default branch?

Reply:
A default case has been added to prevent the code from getting into undefined status.